connect/runtime/src/main/java/org/apache/kafka/connect/runtime/rest/admin/LoggingResource.java

```diff
@@ -0,0 +1,173 @@
+import org.apache.log4j.Level;
+import org.apache.log4j.LogManager;
+import org.apache.log4j.Logger;
+
+import javax.ws.rs.Consumes;
+import javax.ws.rs.GET;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Enumeration;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Objects;
+import java.util.TreeMap;
+
+/**
+ * A set of endpoints to adjust the log levels of runtime loggers.
+ */
+@Path("/admin/loggers")
+@Produces(MediaType.APPLICATION_JSON)
+@Consumes(MediaType.APPLICATION_JSON)
+public class LoggingResource {
+
+    private static final String ROOT_LOGGER_NAME = "@root";
+
+    /**
+     * List the current loggers that have their levels explicitly set and their log levels.
+     *
+     * @return a list of current loggers and their levels.
+     */
+    @GET
+    @Path("/")
+    public Response listLoggers() {
+        Enumeration en = LogManager.getCurrentLoggers();
+        Map<String, Map<String, String>> loggers = new TreeMap<>();
+        while (en.hasMoreElements()) {
+            Logger current = (Logger) en.nextElement();
+            // exclude loggers that do not have a specific level set
+            if (current.getLevel() != null) {
+                loggers.put(current.getName(), levelToMap(current.getLevel()));
+            }
+        }
+
+        Logger root = LogManager.getRootLogger();
+        if (root.getLevel() != null) {
+            loggers.put(ROOT_LOGGER_NAME, levelToMap(root.getLevel()));
+        }
+
+        return Response.ok(loggers).build();
+    }
+
+    /**
+     * Get the log level of a named logger.
+     *
+     * @param namedLogger name of a logger
+     * @return level of the logger, effective level if the level was not explicitly set.
+     */
+    @GET
+    @Path("/{logger}")
+    public Response getLogger(final @PathParam("logger") String namedLogger) {
+        Objects.requireNonNull(namedLogger, "require non-null name");
+
+        Enumeration en = LogManager.getCurrentLoggers();
+        Logger logger = null;
+        while (en.hasMoreElements()) {
+            Logger l = (Logger) en.nextElement();
+            if (namedLogger.equals(l.getName())) {
+                logger = l;
+                break;
+            }
+        }
+        if (logger == null) {
+            throw new NotFoundException("logger " + namedLogger + " not found.");
+        } else {
+            Map<String, String> levelMap = new HashMap<>();
+
+            String level;
+            if (logger.getLevel() == null) {
+                level = String.valueOf(logger.getEffectiveLevel());
+            } else {
+                level = String.valueOf(logger.getLevel());
+            }
+
+            levelMap.put("level", level);
```

Comment:
Could this be replaced with the following?
```
Map<String, String> levelMap = levelToMap(logger.getLevel());
```

Or, better yet, if `levelToMap(...)` were changed to take a `Level` and encapsulated the logic of whether the level was set, then we could clean up other places where this method is used and this code:
```
Map<String, String> levelMap = levelToMap(logger);
```